src/Security/Voter/Committee/VoteCommitteeVoter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace AppBundle\Security\Voter\Committee;
+
+use AppBundle\Committee\CommitteePermissions;
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Committee;
+use AppBundle\Repository\CommitteeMembershipRepository;
+use AppBundle\Security\Voter\AbstractAdherentVoter;
+
+class VoteCommitteeVoter extends AbstractAdherentVoter
+{
+    private $repository;
+
+    public function __construct(CommitteeMembershipRepository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    protected function supports($attribute, $subject)
+    {
+        return $subject instanceof Committee
+            && CommitteePermissions::VOTE === $attribute
+        ;
+    }
+
+    /**
+     * @param Committee $committee
+     */
+    protected function doVoteOnAttribute(string $attribute, Adherent $adherent, $committee): bool
+    {
+        if (!$committee->isApproved()) {
+            return false;
+        }
+
+        $membership = $adherent->getMembershipFor($committee);
+
+        if (CommitteePermissions::VOTE === $attribute) {
```

Comment:
this check is already done in the `supports` method.

if the attribute isn't CommitteePermissions::VOTE : then this voter won't even be called.